<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ScheduleSettingsDialogClass</class>
 <widget class="QDialog" name="ScheduleSettingsDialogClass">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>830</width>
    <height>638</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>610</width>
    <height>530</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="tabShape">
      <enum>QTabWidget::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_General">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_7">
       <item row="1" column="0">
        <widget class="QCheckBox" name="cb_LoadStorageNodesOnStartUp">
         <property name="toolTip">
          <string>when selected updates the storage nodes information from the data monitor at startup</string>
         </property>
         <property name="text">
          <string>Load storage nodes from Data Monitor on start up</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QCheckBox" name="cb_LoadDefaultSettings">
         <property name="toolTip">
          <string>when selected the default scheduler settings will be loaded at startup</string>
         </property>
         <property name="text">
          <string>Load previously saved default settings on start up</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_StorageNodes">
      <attribute name="title">
       <string>Storage</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_8">
       <item row="1" column="0" colspan="2">
        <widget class="QTreeWidget" name="treeWidgetStorageNodes">
         <column>
          <property name="text">
           <string notr="true">1</string>
          </property>
         </column>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="pb_RefreshStorageNodesInfo">
         <property name="maximumSize">
          <size>
           <width>60</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>updates the storage nodes info (disk space and status) from the data monitor</string>
         </property>
         <property name="text">
          <string>Refresh</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_SAS">
      <attribute name="title">
       <string>SAS</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>SAS host name:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="edit_SAShostname">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>22</height>
          </size>
         </property>
         <property name="statusTip">
          <string>the host that holds the SAS database</string>
         </property>
         <property name="whatsThis">
          <string>The name or IP-address of the host on which the SAS database is running.</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Database name:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="edit_SASdatabase">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>22</height>
          </size>
         </property>
         <property name="statusTip">
          <string>the name of the database</string>
         </property>
         <property name="whatsThis">
          <string>The name of the SAS database itself.</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>SAS user name:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="edit_SASusername">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>22</height>
          </size>
         </property>
         <property name="statusTip">
          <string>the internal database user that has write rights </string>
         </property>
         <property name="whatsThis">
          <string>The user name to get access to the database.</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>SAS password:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="edit_SASpassword">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>22</height>
          </size>
         </property>
         <property name="statusTip">
          <string>password for the internal user of the database</string>
         </property>
         <property name="whatsThis">
          <string>The password belonging to the SAS database account.</string>
         </property>
         <property name="echoMode">
          <enum>QLineEdit::Password</enum>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="pushButtonTestSASConnection">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Test connection</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_DataMonitor">
      <attribute name="title">
       <string>Data Monitor</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>DM host name:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="edit_DMhostname">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>22</height>
          </size>
         </property>
         <property name="statusTip">
          <string>the host name of the machine that holds the data monitor database</string>
         </property>
         <property name="whatsThis">
          <string>The name or IP-address of the host on which the Data Monitor database is running.</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>Database name:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="edit_DMdatabase">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>22</height>
          </size>
         </property>
         <property name="statusTip">
          <string>the database name of the data monitor</string>
         </property>
         <property name="whatsThis">
          <string>The name of the Data Monitor database itself.</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_16">
         <property name="text">
          <string>DM user name:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="edit_DMusername">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>22</height>
          </size>
         </property>
         <property name="statusTip">
          <string>the data monitor user name</string>
         </property>
         <property name="whatsThis">
          <string>The user name to get access to the database.</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>DM password:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="edit_DMpassword">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>22</height>
          </size>
         </property>
         <property name="statusTip">
          <string>the data monitor user password</string>
         </property>
         <property name="whatsThis">
          <string>The password belonging to the Data Monitor database account.</string>
         </property>
         <property name="echoMode">
          <enum>QLineEdit::Password</enum>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="pushButtonTestDataMonitorConnection">
         <property name="text">
          <string>Test connection</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_Deletion">
      <attribute name="title">
       <string>Permanent delete filter</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Directory exclude strings:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QTreeWidget" name="treeWidgetExcludeStrings">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Provide a list of strings to prevent deleting directories that contain the string from any deletion action.&lt;/p&gt;&lt;p&gt;If such a string is encountered in the directory name during a deletion action then that directory will not be deleted if the checkbox is checked.&lt;/p&gt;&lt;p&gt;You can temporarily switch of the filter by unchecking the checkbox. This filter will then be disabled so that these directies can be deleted. When a filter is disabled you won't get a warning about such a deletion.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Add a smart description to each filter string for your fellow science-supporters!&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Choose Save Settings from the Settings menu to save the filters for future cleanup runs.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;(Double-click or press a key on an filter item to edit it. Press delete key or empty the string to delete the item.)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::AnyKeyPressed|QAbstractItemView::DoubleClicked|QAbstractItemView::EditKeyPressed</set>
         </property>
         <column>
          <property name="text">
           <string notr="true">1</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pb_Cancel">
       <property name="maximumSize">
        <size>
         <width>70</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pb_Ok">
       <property name="maximumSize">
        <size>
         <width>70</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>cb_LoadDefaultSettings</tabstop>
  <tabstop>pb_Cancel</tabstop>
  <tabstop>pb_Ok</tabstop>
  <tabstop>tabWidget</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pushButtonTestSASConnection</sender>
   <signal>clicked()</signal>
   <receiver>ScheduleSettingsDialogClass</receiver>
   <slot>testSASconnection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>88</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>today()</slot>
  <slot>scheduleStartDateChanged(QDate)</slot>
  <slot>scheduleDurationChanged(int)</slot>
  <slot>scheduleEndDateChanged(QDate)</slot>
  <slot>testSASconnection()</slot>
 </slots>
</ui>
